docs/tutorials/gitflow.md

```diff
@@ -0,0 +1,247 @@
+
+- add search functionality
+- display a picture along with articles
+- fix a bug
+
+Finishing those objectives will constitute a release. A feature branch will be created for each of the features (surprise!).
+
+![creating new branches to work on a feature](../images/fig2.jpg "Title")
+
+So, each branch gives a developer an isolated workspace in which to make changes and add code, without breaking the whole system.
+
+![alt text](../images/fig3.jpg "Title")
+
+The feature is finished once the changes have been reviewed and tested (a process explained later on in  detail), we will merge that branch back into the develop branch.
+
+![alt text](../images/fig4.jpg "Title")
+
+This process will continue for each branch, until we're finished with each feature and have merged it back into the develop branch.
+
+![alt text](../images/fig5.jpg "Title")
+
+And finally, once we've added the features we wanted, and we're sure that our develop branch is working like its supposed to, we can create a release.
+
+![alt text](../images/fig6.jpg "Title")
+
+This is an overview of how the process works. In the next section, I'll go more in depth about the commands you'll use, and how you'll be working with a local repo on your machine and communicating with the repos on the GitHub servers.
+
+## The Detailed Process
+
+The first thing we need is a repository to work on. This will already be created on GitHub under the Praekelt or Unicore Organizations. You will the clone the repo using
+
+```
+$ git clone <REPO SSH URL>
+```
+
+So for example
+
+```
+$ git clone git@github.com:universalcore/springboard.git
+```
+
+Congrats! You've now got a local version of the code that's hosted on the GitHub servers. It looks something like this:
+
+![alt text](../images/fig7.jpg "Title")
+
+Note that you only have the develop branch on your machine. This is because it will only clone the default branch from github. You need to set up a local version of the master branch and then use gitflow (make sure it's installed) to take care of some of stuff behind the scenes, which you don't need to worry about.
+
+```
+$ cd <repo name>
+$ git checkout master
+```
+
+This will automatically create a local master branch and switch you to the master branch, switch back to the develop branch and use Gitflow to work its magic
+
+```
+$ git checkout develop
+$ git flow init
+```
+
+You'll get a bunch of options that you need to confirm that looks like this:
+```
+Branch name for production releases: [master]
+Branch name for "next release" development: [develop]
+Feature branches? [feature/]
+Release branches? [release/]
+Hotfix branches? [hotfix/]
+```
+
+Simply accept the defaults and stuff has happened!
+
+![alt text](../images/fig8.jpg "Title")
+
+Next, you need to create and start up your virtualenv, which is explained [in a later article](/wow/2015/06/10/virtualenv.html).
```

Comment:
Broken link here, I think should be something like `[in another article](/tutorials/virtualenv.html)`